src/shapes/IText/ITextBehavior.ts

```diff
@@ -336,14 +340,13 @@ export abstract class ITextBehavior<
    */
   selectWord(selectionStart: number) {
     selectionStart = selectionStart || this.selectionStart;
-    const newSelectionStart = this.searchWordBoundary(
-        selectionStart,
-        -1
-      ) /* search backwards */,
-      newSelectionEnd = this.searchWordBoundary(
-        selectionStart,
-        1
-      ); /* search forward */
+    // search backwards
+    const newSelectionStart = this.searchWordBoundary(selectionStart, -1),
+      // search forward
+      newSelectionEnd = Math.max(
+        newSelectionStart,
+        this.searchWordBoundary(selectionStart, 1)
+      );
```

Comment:
added a safeguard

Reply:
because it move back if it encounters \n and then if that is the only char on line it does move forward
The test proves it